src/main/java/com/thealgorithms/scheduling/HighestResponseRatioNextScheduling.java

```diff
@@ -0,0 +1,140 @@
+            finishedProcessCount++;
+        }
+
+        return turnAroundTime;
+    }
+
+    /**
+     * Calculates the Waiting Time (WT) for each process.
+     *
+     * @param turnAroundTime The Turn Around Times for each process.
+     * @param burstTimes     The burst times for each process.
+     * @return An array of Waiting Times for each process.
+     */
+    public static int[] calculateWaitingTime(int[] turnAroundTime, int[] burstTimes) {
+        int[] waitingTime = new int[turnAroundTime.length];
+        for (int i = 0; i < turnAroundTime.length; i++) {
+            waitingTime[i] = turnAroundTime[i] - burstTimes[i];
+        }
+        return waitingTime;
+    }
+
+    /**
+     * Finds the next process to be scheduled based on arrival times and the current time.
+     *
+     * @param arrivalTimes    Array of arrival times for each process.
+     * @param finishedProcess Array indicating whether each process has finished.
+     * @param currentTime     The current time in the scheduling process.
+     * @return The index of the next process to be scheduled.
+     */
+    private static int findNextProcess(int[] arrivalTimes, boolean[] finishedProcess, int currentTime) {
+        for (int i = 0; i < arrivalTimes.length; i++) {
+            if (!finishedProcess[i] && arrivalTimes[i] <= currentTime) {
+                return i;
+            }
+        }
+        return 0;
+    }
+
+    /**
+     * Finds the process with the highest response ratio.
+     *
+     * @param processNames    Array of process names.
+     * @param arrivalTimes    Array of arrival times for each process.
+     * @param burstTimes      Array of burst times for each process.
+     * @param finishedProcess Array indicating whether each process has finished.
+     * @param currentTime     The current time in the scheduling process.
+     * @return The index of the process with the highest response ratio.
+     */
+    private static int findHighestResponseRatio(String[] processNames, int[] arrivalTimes, int[] burstTimes, boolean[] finishedProcess, int currentTime) {
+        double maxResponseRatio = -1;
+        int loc = -1;
+
+        for (int i = 0; i < processNames.length; i++) {
+            if (!finishedProcess[i] && arrivalTimes[i] <= currentTime) {
+                double responseRatio = (double) (burstTimes[i] + currentTime - arrivalTimes[i]) / burstTimes[i];
+                if (responseRatio > maxResponseRatio) {
+                    maxResponseRatio = responseRatio;
+                    loc = i;
+                }
+            }
+        }
+        return loc;
+    }
+
+    /**
+     * Sorts the indices of the arrival times array in ascending order.
+     *
+     * @param arrivalTimes Array of arrival times for each process.
+     * @return An array of indices sorted by the corresponding arrival times.
+     */
+    private static int[] sortIndicesByArrival(int[] arrivalTimes) {
+        Integer[] indices = new Integer[arrivalTimes.length];
```

Comment:
Better to use `int[]` instead of `Integer[]` to avoid  unnecessary boxing/unboxing. Since method return `int[]`.

Or maybe it can be better rewritten to kind of such implementation:

```
private static int[] sortIndicesByArrival(int[] arrivalTimes) {
    return IntStream.range(0, arrivalTimes.length)
        .boxed()
        .sorted(Comparator.comparingInt(i -> arrivalTimes[i]))
        .mapToInt(Integer::intValue)
        .toArray();
}
```

What do you think?

Reply:
Insights like these is why I like contributing to open source and learning valuable information!
I simplified the function call to the following:

```java
 public static int[] calculateTurnAroundTime(final String[] processNames, final int[] arrivalTimes, final int[] burstTimes, final int noOfProcesses) {
	...
	Arrays.sort(processes, Comparator.comparingInt(p -> p.arrivalTime));
	...;
```